mne/source_estimate.py

```diff
@@ -1894,18 +1894,26 @@ def save(self, fname, ftype="stc", *, overwrite=False, verbose=None):
             spaces are obtained by adding ``"-lh.stc"`` and ``"-rh.stc"`` (or
             ``"-lh.w"`` and ``"-rh.w"``) to the stem provided, for the left and
             the right hemisphere, respectively.
-        ftype : str
-            File format to use. Allowed values are ``"stc"`` (default),
-            ``"w"``, and ``"h5"``. The ``"w"`` format only supports a single
-            time point.
+        ftype : str | None
+            File format to use. If None, the file format is inferred from the
+            file extension. Allowed values are ``"stc"``, ``"w"``, and ``"h5"``.
+            The ``"w"`` format only supports a single time point.
         %(overwrite)s
 
             .. versionadded:: 1.0
         %(verbose)s
         """
         fname = str(_check_fname(fname=fname, overwrite=True))  # checked below
+        if ftype is None:
+            if fname.endswith((".stc", "-lh.stc", "-rh.stc")):
+                ftype = "stc"
+            elif fname.endswith((".w", "-lh.w", "-rh.w")):
+                ftype = "w"
+            elif fname.endswith(".h5"):
+                ftype = "h5"
+            else:
+                raise ValueError("Cannot infer file type, please specify ftype.")
```

Comment:
the suggestion in the linked issue was to fall back to `"stc"` if the file type could not be inferred (not raise an error).

Reply:
```suggestion
            else:
                logger.info(
                    "Cannot infer file type from `fname`; falling back to `.stc` format"
                )
                ftype = "stc"
```